Simplified/NYPLReadiumViewSyncManager.m

```diff
@@ -111,23 +112,28 @@ - (void)syncReadingPositionForBook:(NSString *)bookID
 {
   __weak NYPLReadiumViewSyncManager *const weakSelf = self;
 
-  [NYPLAnnotations syncReadingPositionOfBook:bookID toURL:URL completionHandler:^(NSDictionary * _Nullable responseObject) {
```

Comment:
the completion callback was using a loose object to pass along the CFI and a few other things, but the parsing of the api response was identical to parsing a bookmark. So we might as well parse a bookmark object and return that instead, and this way reuse the bookmark parsing code as-is.